src/dataviews/histogram-dataview-model.js

```diff
@@ -135,6 +135,12 @@ module.exports = DataviewModelBase.extend({
         this.trigger('histogram_sizes', this);
       }
     }, this);
+  },
+
+  _refreshAndResetFilter: function () {
+    this.refresh();
```

Comment:
[`refresh` is actually just fetching](https://github.com/CartoDB/cartodb.js/blob/v4/src/dataviews/dataview-model-base.js#L160-L162) so perhaps we could use `fetch` here.